gems/aws-sdk-core/lib/aws-sdk-core/endpoints.rb

```diff
@@ -33,8 +33,60 @@ def resolve_auth_scheme(context, endpoint)
 
       private
 
+      def merge_signing_defaults(auth_scheme, config)
+        if %w[sigv4 sigv4a].include?(auth_scheme['name'])
+          auth_scheme['signingName'] ||= sigv4_name(config)
+          if auth_scheme['name'] == 'sigv4a'
+            auth_scheme['signingRegionSet'] ||= config.sigv4a_signing_region_set
```

Comment:
I think this order should be different and the configured signing_region_set should act as an override.  Someething like `config.sigv4a_signing_region_set || auth_scheme['signingRegionSet'] || cfg.region`

Reply:
Yes. This is old code.